### PR TITLE
created event-schema scaffold

### PR DESCRIPTION
**Pull request by anyelopetit**
### Description
An EventSchema establish a default config for each event in an App so that each external event will be processed based on the config.

## Columns

```ruby
name: # Name that must be unique.
description: # A short description
schema: # The schema to be received will be established,
        # this will allow validating the payload in a certain way. 
creator: # User who creates the event.
app: # Associated app.
```

## Schema

It will be divided into two sections

```{ data: [...], recipient: [...] }```

```ruby
data: [...] # It must be a column name list
recipient: [...]  # It must be a column name list
```

## Request's example

Let's imagine that we want to send a notification to several users, these users are in charge of processing a purchase order.

```javascript
curl --location --request POST 'http://localhost:3000/api/v1/apps/:id/event_schemas' \
--header 'Content-Type: application/json' \
--data-raw '{ 
  "name": "created_order",
  "description": "Notification for admins",
  "schema": {
    "data": ["order_id", "order_name", "order_price", "product_image"]
    "recipient": ["email", "user_name"]
  }
}'
```

### CODE REVIEW

File: spec/controllers/v1/apps_controller_spec.rb [hunk 1]
```diff
@@ -6,14 +6,14 @@
   context "without a authenticated user" do
     describe "GET #index" do
       it "401 - Unauthorized" do
```
Comment (gbrlmrllo) [inline spec/controllers/v1/apps_controller_spec.rb:8]: 👍 thanks for this fix.

File: spec/controllers/v1/apps_controller_spec.rb [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
-        get :show, params: { id: "id" }
+        get :index
         expect(response.status).to eq(401)
       end
     end
 
     describe "GET #show" do
       it "401 - Unauthorized" do
-        get :index
+        get :show, params: { id: "id" }
```
Comment (gbrlmrllo) [inline spec/controllers/v1/apps_controller_spec.rb:16]: 👍

File: app/models/event_schema.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# An EventSchema establish a default config for each event so that each
+# external events will be processed based on the config.
+
+# For more information, see https://github.com/gbrlmrllo/chatops_builder_api/issues/11
+class EventSchema < ApplicationRecord
+  belongs_to :creator, class_name: "User", optional: false
+  belongs_to :app
```
Comment (gbrlmrllo) [inline app/models/event_schema.rb]: It should be required.

```suggestion
  belongs_to :app, optional: false
```

File: app/models/event_schema.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  # has_many :events # TODO
+  # has_many :integrations # TODO
+
+  validates :name, presence: true, uniqueness: { scope: :creator_id }
+
+  scope :schema_structure, -> { { data: [], recipient: [] } }
```
Comment (gbrlmrllo) [inline app/models/event_schema.rb]: Scope methods are generally used to define queries, it is confusing to see it and that it does not implement an ActiveRecord query, it is preferable to declare this as a class method.

```ruby 
def self.schema_structure
   { data: [], recipient: [] }
end
```

Should also add a spec for this method.

File: app/controllers/v1/event_schemas_controller.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module V1
+  class EventSchemasController < V1::AppsController
+    before_action :authenticate_user!
+    before_action :set_app
+    before_action :set_event_schema, only: %i[show update destroy]
+
+    # GET api/v1/apps/1/event_schemas
```
Comment (gbrlmrllo) [inline app/controllers/v1/event_schemas_controller.rb]: The comment is a little more informative if we know what is the key of the id.

```# GET api/v1/apps/:app_id/event_schemas```

File: app/models/app.rb [hunk 6]
```diff
@@ -3,6 +3,7 @@
 class App < ApplicationRecord
   belongs_to :owner, class_name: "User", optional: false
   has_one :credential, dependent: :destroy
+  has_many :event_schemas, dependent: :destroy
```
Comment (gbrlmrllo) [inline app/models/app.rb:6]: Can you add a spec for this in `model/app_spec.rb`?

**Review by gbrlmrllo — CHANGES_REQUESTED** [review]
@anyelopetit Excellent job 💯everything looks fine, code wise. 🥇 

I left a few minor comments.  💬

File: spec/controllers/v1/apps_controller_spec.rb (hunk 1 above)
Comment (anyelopetit) [reply spec/controllers/v1/apps_controller_spec.rb:8]: you're welcome!

File: spec/models/event_schema_spec.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    subject { build(:event_schema, creator: create(:user)) }
+
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_uniqueness_of(:name).scoped_to(:creator_id) }
+  end
+
+  describe ".schema_structure" do
+    let(:schema_structure) { described_class.schema_structure }
+
+    it "is a Hash" do
```
Comment (gbrlmrllo) [inline spec/models/event_schema_spec.rb]:
```suggestion
    it "returns a hash" do
```

File: spec/models/event_schema_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      expect(schema_structure.class).to eq(Hash)
+    end
+
+    it "has only two keys" do
```
Comment (gbrlmrllo) [inline spec/models/event_schema_spec.rb]: This test and the one below are redundant, you can remove this one.

**Review by gbrlmrllo — CHANGES_REQUESTED** [review]

**Pull request merged**